### ISSUE TITLE
./rrobbes/smallcompanies.md

### ISSUE THREAD

**Issue by timm** [ds4se/chapters#70]
After review, relabel to 'reviewTwo'. After second review, relabel to 'EditorsComment'.

**Comment by prechelt** [comment]
# Review template

Before filling in this review, please read our [Advice to Reviewers](how2review.md).

_(If you have confidential comments about this chapter, please email them to one of the book editors._)

## Title of chapter

Software Analytics for Small Software Companies: More Questions than Answers

## URL to the chapter

https://github.com/ds4se/chapters/blob/master/rrobbes/smallcompanies.md

## Message?

What is the chapter's  clear and approachable take away message?

Researchers! Make sure to develop techniques relevant for 
small companies as well; they have different needs.

## Accessible?

Is the chapters written for a generalist audience (no excessive use of technical terminology) with a minimum of diagrams and references?

I am not sure what a "generalist" audience is intended to be (see
discussion below), but the chapter is not very technical in any case.

## Size?

Is the chapter the right length?
Should anything missing be added?
Can anything superfluous be removed (e.g. by deleting some section that does not work so well or  by using less jargon, less formulae, lees diagrams, less references).?
What are the aspects of the chapter that authors  SHOULD  change?
- The introduction reads like an ICSE submission, trying to sell a topic
  to hyper-critical reviewers. In our case, readers from large companies
  will simply skip this chapter and readers from small ones will be interested
  no matter whether there are many other small companies or not.
  I suggest to shorten this to simply state that so far most work on 
  analytics assumed a large company and small companies need different advice.
- "Projects taken by small software companies [...]"
  I first thought this means data analytics projects, but in fact
  it means software development projects. 
  I suggest to prepend the word "software".
- "Taking the example of Chile, we know that small software companies suffer 
  from a high mortality. Certainly, analytics could be helpful there."
  Hmm. At this spot I wondered whether this chapter is going to be 
  _about_ small companies or _for_ them?
- "Software Analytics approaches should consider the cases were little data
  is available"
  Oh, wow. I now recognize that my first comment made an assumption not
  shared by the author of this chapter.
  I assumed this should be a book for data analytics practitioners.
  Romain assumes it is one for data analytics researchers.
  DEAR EDITORS, WE NEED A STATEMENT FROM YOU ON THIS: WHAT IS INTENDED?
  (Maybe one could split the whole book into two parts, one for each group.)
- The references currently all form one single paragraph

## Gotta Mantra?

Care for Squirrels, not only Elephants: Software Analytics for Small Companies

## Best Points

What are the best points of the chapter that the authors should NOT change?

This chapter follows a different mission than I would have thought:
Not providing advice for small-company practitioners, rather
advice for researchers how to cater for small companies as well.
It does a good job at that.

**Comment by timm** [comment]
@prechelt : so if your comment that the paper could be adjusted to better suite (e.g.) the IT manager of a small company who might be thinking "i won't use analytics since my company is too small"?

**Comment by prechelt** [comment]
@timm : No. The chapter talks mostly to researchers, but intelligent practitioners will be able to take home the message, too. 
The question is: Does the book have only one audience (researchers or practitioners) in mind? Then which? Or both? Or are they intended to be one and the same? That, I think, does _not_ work.

**Comment by timm** [comment]
well, i'm sure @tzimmermsr @lauriew have a view but mine is that the audience of the book is BROAD. where "BROAD" means the user community or  the manager community or the the developer community

**Comment by lauriew** [comment]
I agree … perhaps including a book that can be used in a seminar class of graduate students. 

Laurei

> On Nov 6, 2015, at 2:21 PM, Tim Menzies notifications@github.com wrote:
> 
> well, i'm sure @tzimmermsr https://github.com/tzimmermsr @lauriew https://github.com/lauriew have a view but mine is that the audience of the book is BROAD. where "BROAD" means the user community or the manager community or the the developer community
> 
> —
> Reply to this email directly or view it on GitHub https://github.com/ds4se/chapters/issues/70#issuecomment-154507624.

---

Laurie Williams, PhD
North Carolina State University
Associate Department Head and Professor, Department of Computer Science
890 Oval Drive, Room 3272
Campus Box 8206
Raleigh, NC 27695-8206

http://collaboration.csc.ncsu.edu/laurie/

**Comment by turhanb** [comment]
## Title of chapter

Software Analytics for Small Software Companies: More Questions than Answers

## URL to the chapter

https://github.com/ds4se/chapters/blob/master/rrobbes/smallcompanies.md

## Message?

Software analytics approaches need scaling "down", driven by unique priorities of small software companies.

## Accessible?

This chapter lists points that are all valid, but it gives the impression that it's written towards our research community in the form of a position/vision statement. I do believe that the main content can be tweaked with a little effort to address practitioners. A few suggestions in the next part.

## Size?
- Sentences like "Needless to say, helping large software companies is a very useful goal. " certainly targets researchers and should be candidates for removal/ reformulation throughout. 
- The three sections from the beginning until "Different goals and needs" indeed try to motivate the difference between small and large companies from software analytics perspective. A more concise approach would be useful to pinpoint those key points, as the focus is expected to be in an example or take-away messages instead of motivating the need. As a result the corresponding part could be significantly shorter and to the point.
- Then the section titled "Different goals and needs" can be refocused to the specific Chilean case with more details; finishing with advice to "data scientists" on what to consider when engaging in data science projects with small companies (e.g. fine-grained data collection for technical and an understanding of priorities for coarse-grained data), as well as advice for small companies (e.g. process on following higher level tasks).
- The logical coupling example could be too a technical term with which some readers might be unfamiliar.

## Gotta Mantra?

One size does not fit all (?)

## Best Points

The Chilean case is certainly interesting and can be elaborated, even the chapter can be centred around that.

**Comment by tzimmermsr** [comment]
@rrobbes Please prepare a new version of your paper by January 13 taking the reviewers' feedback into account. They offer great advice on how to improve the chapter.

My challenge to you is to make it more appealing to a broad audience. Don't focus so much on the ICSE reviewer as a reader, but think about what a software engineer in a small (or even large) company could take away from the chapter. 

Reduce the number of references where possible. Those are great for ICSE reviewers but make the chapter more complicated to read for a broad audience.

**Comment by rrobbes** [comment]
Thanks for the comments! I revised my chapter, cutting half of the references and of the intro, refocusing it on the case of Amisoft, and expanding the discussion of Codealike towards the end of the paper. Sorry for the delay!

**Comment by tzimmermsr** [comment]
Thanks @rrobbes. This looks good to go from my side. One change request though:

Please change

"Software analytics have been shown to be useful for many use cases. Examples are numerous, and many of these are mentioned in this book. Software analytics has also been successfully applied in the industry. Again, the examples are numerous. These examples share one characteristic: they come mostly from large software companies."

to

"Software analytics has been shown to be useful for many applications and many examples are mentioned in this book. Software analytics is successfully applied in industry, especially in large software companies."

Why? Currently, the intro suggests that there are two worlds: a non-industry world that showed the value of analytics and an industry-world that applied analytics. I'd prefer to have it both as one world.
